### PR TITLE
Added package gulp task.

### PR DESCRIPTION
**Pull request by dlebu**
Added a new gulp task: `gulp package`

### CODE REVIEW

File: tools/gulp-extras.js [hunk 1]
```diff
@@ -99,7 +100,24 @@ var checkImports = function() {
...
+
+    process.stdout.on('data', function(data) {
+        console.log("" + data);
+    });
+
+    process.stderr.on('data', function(data) {
+        console.error("" + data);
+    });
+
+    process.on('exit', function(code) {
```
Comment (hamiltonia) [inline tools/gulp-extras.js]: should you also handle 'error'?

**Comment by hamiltonia** [comment]
:+1:

File: tools/gulp-extras.js [hunk 2]
```diff
@@ -99,7 +100,24 @@ var checkImports = function() {
...
+        callback(code === 0 ? undefined : "Error code: " + code);
```
Comment (digeff) [inline tools/gulp-extras.js]: Although basically identically, I'd do:
code === 0 ? callback() : callback("Error code: " + code);

**Comment by joshuaskelton** [comment]
:+1:

File: tools/gulp-extras.js (hunk 1 above)
Comment (dlebu) [inline tools/gulp-extras.js]: The exit code will take care of reporting errors to gulp via the `callback`, while the error output will be   streamed to the console via `process.stderr`. To my knowledge, there is no `error` event on process.

Comment (dlebu) [inline tools/gulp-extras.js]: But there is one on `child_process` which is what we are using here. :) Let me take care of that.

**Comment by dlebu** [comment]
Thanks everyone, merged to master.

**Pull request merged**